mdal/mdal_utils.cpp

```diff
@@ -1041,8 +1052,34 @@ bool MDAL::Library::loadLibrary()
   d->mLibrary = LoadLibrary( d->mLibraryFile.c_str() );
   SetErrorMode( uOldErrorMode );
 
+#if 1 //for debugging
+  DWORD errorMessageID = GetLastError();
+  if ( d->mLibrary )
+  {
+    OutputDebugString( "++++++++++++++++++++++++++++++  load lib success for:" );  OutputDebugString( d->mLibraryFile.c_str() ); OutputDebugString( "\n" );
+  }
+  else
+  {
+    OutputDebugString( "------------------------------  load lib fail for:" );  OutputDebugString( d->mLibraryFile.c_str() ); OutputDebugString( "\n" );
+    std::string message;
+
+    if ( errorMessageID != 0 )
+    {
+
+      LPSTR messageBuffer = nullptr;
+      size_t size = FormatMessage( FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS,
+                                   NULL, errorMessageID, MAKELANGID( LANG_NEUTRAL, SUBLANG_DEFAULT ), ( LPSTR )&messageBuffer, 0, NULL );
+
+      std::string message( messageBuffer, size );
+    }
+    OutputDebugString( "------------------------------  code error:" );  OutputDebugString( std::to_string( errorMessageID ).c_str() ); OutputDebugString( "\n" );
+    OutputDebugString( "------------------------------  message:" );  OutputDebugString( message.c_str() ); OutputDebugString( "\n" );
+  }
+#endif
```

Comment:
I will create a new function from this `void debug_win_loading` and ifdef it to `if defined(windows) and defined(debug)` or so.?

Reply:
or just remove it?

Reply:
maybe remove all this stuff once the implementation will be ok. For now, it is convenient in run time when testing...